app/code/Magento/Store/App/Response/Redirect.php

```diff
@@ -63,6 +70,7 @@ public function __construct(
         \Magento\Framework\Encryption\UrlCoder $urlCoder,
         \Magento\Framework\Session\SessionManagerInterface $session,
         \Magento\Framework\Session\SidResolverInterface $sidResolver,
+        \Magento\Backend\App\Area\FrontNameResolver $backendFrontNameResolver,
```

Comment:
Note that `Backend` module must not be referenced in `Store` module.

Reply:
Ah, I just saw this comment now. I was struggling to find a method that would return the actual full admin url is it has been customized. There are some methods in some deprecated helpers.

https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Backend/App/Area/FrontNameResolver.php#L103

Essentially, what needs to be done is to get this backendUrl inside of this Redirect class. 

My further commits are probably not going to cut it if the Backend module cannot be referenced from here.